core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2617,6 +2619,17 @@ private[spark] object Utils extends Logging {
     redact(redactionPattern, kvs.toArray)
   }
 
+  def redactCommandLineArgs(conf: SparkConf, commands: Seq[String]): Seq[String] = {
+    val redactionPattern = conf.get(SECRET_REDACTION_PATTERN)
+    commands.map {
+      case PATTERN_FOR_COMMAND_LINE_ARG(key, value) =>
+        val (newKey, newValue) = redact(redactionPattern, Seq((key, value))).head
```

Comment:
While I assign key as newKey, actually `key` will not be reacted. Please refer `def redact(redactionPattern: Regex, kvs: Seq[(String, String)])` in this class.

https://github.com/apache/spark/blob/4808393449ccad2c6bc73c91d0ed8dd8f60c7054/core/src/main/scala/org/apache/spark/util/Utils.scala#L2584-L2604

I'd just like to rely on the return value for that method to fully leverage the method in case of further modification on that method.

Reply:
How about making that clear by ignoring the key in the return value, then?

Reply:
OK. That would keep the intention from caller. Will address.